builder/main.py

```
@@ -1270,276 +1270,276 @@ def _download_partition_image(env, fs_type_filter=None):
+    sector_size = int.from_bytes(fs_data[0x0B:0x0D], byteorder='little')
 
-        # Read sector size from FAT boot sector (offset 0x0B, 2 bytes, little-endian)
-        sector_size = int.from_bytes(fs_data[0x0B:0x0D], byteorder='little')
-        # print(f"  Sector size from boot sector: {sector_size} bytes")
+    # Validate sector size
+    if sector_size not in [512, 1024, 2048, 4096]:
+        print(f"Error: Invalid sector size {sector_size}. Must be 512, 1024, 2048, or 4096")
+        return 1
+
+    # Mount with fatfs-python
+    from fatfs import RamDisk, create_extended_partition
```

Ruff (0.14.10)

Redefinition of unused `create_extended_partition` from line 25

Remove definition: `create_extended_partition`

(F811)